libsquoosh/src/index.js

```diff
@@ -66,15 +77,15 @@ async function encodeImage({
     const optionToOptimize = encoders[encName].autoOptimize.option;
     const decoder = await encoders[encName].dec();
     const encode = (bitmapIn, quality) =>
-      encoder.encode(
+      handleStringError('encode', encoder.encode(
```

Comment:
Let’s not nest these functions calls. That seems confusing.

Reply:
Ah, I agree. Fixed!